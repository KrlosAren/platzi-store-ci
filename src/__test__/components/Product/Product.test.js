import { mount } from 'enzyme';
import React from 'react';
import Product from '../../../components/Product';
import ProviderMock from '../../../__mocks__/providerMocks';

describe('Name of the group', () => {
  const handleAddToCart = jest.fn();
  const wrapper = mount(
    <ProviderMock>
      <Product product={ProviderMock} handleAddToCart={handleAddToCart} />
    </ProviderMock>
  );
  test('should be match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('comprobar el boton de pagar', () => {
    wrapper.find('button').simulate('click');

    expect(handleAddToCart).toHaveBeenCalledTimes(1);
  });
});
