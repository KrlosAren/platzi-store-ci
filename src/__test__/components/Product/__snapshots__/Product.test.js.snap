// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should be match to snapshot 1`] = `
<div
  className="Products-item"
>
  <img />
  <div
    className="Products-item-info"
  >
    <h2>
      <span>
        $
      </span>
    </h2>
    <p />
  </div>
  <button
    type="button"
  >
    Comprar
  </button>
</div>
`;
